src/StoreKeeper/WooCommerce/B2C/Factories/LoggerFactory.php

```diff
@@ -143,9 +143,10 @@ public static function generateMetadata(string $key, \Throwable $exception, arra
             'exception-class' => get_class($exception),
         ];
 
-        $fullMetadata = array_merge($metadata, $customMetadata);
-        $fullMetadata['exception-message'] = $metadata['exception-message'];
+        if (method_exists($exception, 'getReference')) {
```

Comment:
```suggestion
        if ($exception instanceof \StoreKeeper\ApiWrapper\Exception\GeneralException) {
```

Better to check for class than for method existance

Reply:
Changes implemented.